src/cython/sktda/preprocessing.py

```diff
@@ -0,0 +1,274 @@
+        """
+        self.scalers = scalers
+        self.use     = use
+
+    def fit(self, X, y=None):
+        """
+        Fit the DiagramPreprocessor class on a list of persistence diagrams: persistence diagrams are concatenated in a big numpy array, and scalers are fit (by calling their fit() method) on their corresponding coordinates in this big array.
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+            y (n x 1 array): persistence diagram labels (unused).
+        """
+        if self.use:
+            if len(X) == 1:
+                P = X[0]
+            else:
+                P = np.concatenate(X,0)
+            for (indices, scaler) in self.scalers:
+                scaler.fit(P[:,indices])
+        return self
+
+    def transform(self, X):
+        """
+        Apply the DiagramPreprocessor function on the persistence diagrams. The fitted scalers are applied (by calling their transform() method) to their corresponding coordinates in each persistence diagram individually.  
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+
+        Returns:
+            Xfit (list of n x 2 or n x 1 numpy arrays): transformed persistence diagrams.
+        """
+        Xfit = [np.copy(d) for d in X]
+        if self.use:
+            for i in range(len(Xfit)):
+                if Xfit[i].shape[0] > 0:
+                    for (indices, scaler) in self.scalers:
+                        Xfit[i][:,indices] = scaler.transform(Xfit[i][:,indices])
+        return Xfit
+
+class Padding(BaseEstimator, TransformerMixin):
+    """
+    This is a class for padding a list of persistence diagrams with dummy points, so that all persistence diagrams end up with the same number of points.
+    """
+    def __init__(self, use=False):
+        """
+        Constructor for the Padding class.
+
+        Attributes:
+            use (bool): whether to use the class or not (default False).
+        """
+        self.use = use
+
+    def fit(self, X, y=None):
+        """
+        Fit the Padding class on a list of persistence diagrams (this function actually does nothing but is useful when Padding is included in a scikit-learn Pipeline).
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+            y (n x 1 array): persistence diagram labels (unused).
+        """
+        return self
+
+    def transform(self, X):
+        """
+        Add dummy points to each persistence diagram so that they all have the same cardinality. All points are given an additional coordinate indicating if the point was added after padding (0) or already present before (1).  
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+
+        Returns:
+            Xfit (list of n x 3 or n x 2 numpy arrays): padded persistence diagrams.
```

Comment:
Doesn't it cause problems that this transform returns something of a different shape than the rest, which doesn't directly look like a persistence diagram?

Reply:
I agree that this might be confusing (I created this class originally because it is the input format that PersLay expects, I don't know how useful this is for other applications though). I tried to make all other classes compatible, but some are not (like BirthPersistenceTransform, which requires nx2 arrays, or classes in `kernel_methods`). I guess another possibility could be to output two arrays: the padded persistence diagrams, and a boolean array, that would be the corresponding mask, although if we do that, we won't be able to use this class in a sklearn pipeline.

Reply:
Ok, it is just this one class, users (and most of your other classes) can just ignore the last coordinate, it doesn't prevent people who want to from adding a different padding class.
Maybe specify in the doc what the dummy point is (the origin (0,0)?).
If it isn't the case yet, it should be easy to have BirthPersistenceTransform accept such arrays, the question is more if it should return nx2 (drop the extra coordinate) or nx3 (ignore but keep it). Not super important though, one can just use BPT before padding.